### DISCUSSION TITLE
Wix 5 Bootstrapper Installer Reopens After Install/Uninstall Completes

### DISCUSSION THREAD

**Discussion by Bondarenko1990** [wixtoolset/issues#8753] [Questions]
I use Wix 5 WixStandardBootstrapperApplication. In the Bundle I added my WPF application to the chain

```
<Chain>
    <MsiPackage Id="Setup" SourceFile="$(var.MyProject.Setup.TargetPath)"  Compressed="yes" />
</Chain>
```

and I added a custom action in which I install the ImDisk bat file.
```
string installCmd = $"/silent /lang:english /installfolder:\"C:\\Program Files\\ImDisk\" > install.log 2>&1";
var startInfo = new ProcessStartInfo(".....\\ImDisk\\install.bat", installCmd );

startInfo.UseShellExecute = false;
startInfo.CreateNoWindow = true;

var proc = new Process();

proc.StartInfo = startInfo;
proc.Start();
proc.BeginOutputReadLine();
proc.BeginErrorReadLine();
proc.WaitForExit();
```

And after the installation is complete, a new copy of the installer opens for some reason.
![image](https://github.com/user-attachments/assets/01c46073-4409-4045-a1b1-433be970a455)

If I remove silent install ImDisk from CustomAction everything is fine.

Tell me what I'm doing wrong. Why does the installer reopen after the installation is complete? What's interesting is that there is no such problem in Wix 3 WixStandardBootstrapperApplication.

**Comment by bevanweiss** [comment]
There should generally only be a single 'installer' running at any one time.
So it would be recommended to use the Bundle to launch the ImDisk installation.

You haven't shown how you've authored the CustomAction in WiX, so it's possible that you are executing it multiple times.
But, best practise is to not try to run a subsequent installer from within an MSI

**Comment by Bondarenko1990** [comment]
@bevanweiss Thanks for the advice. I run the ImDisk installation only once in CustomAction! I checked more than once...
How can I run the installation of ImDisk in a bundle? I mean "install.bat" file.

**Reply by wmanning** [reply]
> I run the ImDisk installation only once in CustomAction!
What does the ImDisk bat file do roughly?  Does it run one or more EXE files or another MSI file? 

You have this:
```XML
<Chain>
    <MsiPackage Id="Setup" SourceFile="$(var.MyProject.Setup.TargetPath)"  Compressed="yes" />
</Chain>
```
Ideally you'll take the stuff ImDisk does and put it in a MSI file as that would be much more reliable.
If that is too much work maybe you can wrap it in an EXE file and run it like this:
```XML
<Chain>
    <MsiPackage Id="Setup" SourceFile="$(var.MyProject.Setup.TargetPath)"  Compressed="yes" />
    <ExePackage SourceFile="some-path/ImDisk-Installer.exe" />
</Chain>
```
See: https://wixtoolset.org/docs/schema/wxs/exepackagepayload/

**Reply by Bondarenko1990** [reply]
Thanks for the advice. I did everything as you said, packed the file into an .exe file, but the result is the same. 
As a result, after the installation is complete, for some reason a new copy of the installer opens.

**Reply by wmanning** [reply]
Strange

And you do have the below element (with maybe different attributes) in your burn .wxs file?
```XML
<BootstrapperApplication>
      <bal:WixStandardBootstrapperApplication LicenseUrl="" SuppressOptionsUI="yes"  Theme="hyperlinkLicense" />
</BootstrapperApplication>
```

It could be returning an error code from the exe package you created so you could add a `<ExitCode>` element under it.  If the imdisk returns error codes.
https://wixtoolset.org/docs/schema/wxs/exitcode/
But I don't think this would cause your issue.
